### PR TITLE
Feat/routing

### PR DESCRIPTION
**Pull request by MihlaliMgwebi**
Hi everyone, I messed up my merge from develop into this branch so it seems like I changed 72 files but that is not the case: I have changed 15 files as follows:

-[X] src/app/app-routing.module.ts
-[X] src/app/app.module.ts
-[X] src/app/app.component.html
-[X] src/app/components/itinerary-item-upsert/itinerary-item-upsert.component.html
-[X] src/app/components/itinerary-item-upsert/itinerary-item-upsert.component.ts
-[X] src/app/components/itinerary-item-upsert/itinerary-item-upsert.component.html
-[X] src/app/components/itinerary-item/itinerary-item.component.ts
-[X] src/app/components/nav-bar/nav-bar.component.html
-[X] src/app/components/nav-bar/nav-bar.component.ts
-[X] src/app/components/trip-upsert/trip-upsert.component.html
-[X] src/app/components/trip-upsert/trip-upsert.component.ts
-[X] src/app/components/trip/trip.component.html
-[X] src/app/components/trip/trip.component.ts
-[X] src/app/store/effects/trip.effects.ts
-[X] src/app/store/effects/user.effects.ts

I will resolve my merge conflicts shortly. ( those were caused by merging my store/setup PR with develop poorly)

### CODE REVIEW

**Review by ntokozo-nhlangulela — APPROVED** [review]

File: src/app/components/itinerary-item/itinerary-item.component.ts [hunk 1]
```diff
@@ -1,12 +1,38 @@
...
+  addItineraryItem(){
+    this.router.navigate([`../trips/${this.selectedCorrelatedTrip?.trip._id}/itinerary-items/add`], { relativeTo: this.route })
+  }
+
+  // UPDATE
+  editItineraryItem(itineraryItem: ItineraryItem){
+    this.router.navigate([`..../trips/${this.selectedCorrelatedTrip?.trip._id}/itinerary-items/${itineraryItem._id}/edit`], { relativeTo: this.route })
+  }
+  // DELETE
+  deleteItineraryItem(deletedItineraryItemId: string | null){
```
Comment (ntokozo-nhlangulela) [inline src/app/components/itinerary-item/itinerary-item.component.ts:34]: deletedItineraryItemId: string | null , What happens in a null case?

File: src/app/components/trip-upsert/trip-upsert.component.html [hunk 2]
```diff
@@ -1 +1,4 @@
 <p>trip-upsert works!</p>
+
+<!--<button (click)="onSubmit()">Save Changes</button>-->
```
Comment (ntokozo-nhlangulela) [inline src/app/components/trip-upsert/trip-upsert.component.html:3]: please remove commented button

File: src/app/components/trip-upsert/trip-upsert.component.ts [hunk 3]
```diff
@@ -1,18 +1,28 @@
...
     this.tripStore.dispatch(upsertTrip({ trip: upsertTripForm.value }));
   }
+  goToTrips() {
+    this.router.navigate(['../../trips'], { relativeTo: this.route });
+  }
+  ngOnInit(): void {
+    this.route.queryParams.subscribe(params => {
+      this.tripId = params['id'];
+    });
+    // console.log('tripId', this.tripId)
```
Comment (ntokozo-nhlangulela) [inline src/app/components/trip-upsert/trip-upsert.component.ts:26]: Remove commented code

File: src/app/components/trip-upsert/trip-upsert.component.html (hunk 2 above)
Comment (MihlaliMgwebi) [reply src/app/components/trip-upsert/trip-upsert.component.html:3]: Wiil do thank you!

File: src/app/components/itinerary-item/itinerary-item.component.ts (hunk 1 above)
Comment (MihlaliMgwebi) [reply src/app/components/itinerary-item/itinerary-item.component.ts:34]: If null it won't delete an item. I will make sure to add user feedback in a separate branch. Thank you for bringing this up.

**Pull request closed without merging**